### PR TITLE
Adds Screechisol

### PR DESCRIPTION
**Pull request by PopNotes**
:cl: PopNotes
add: Adds Screechisol, a toxin that makes people scream based on how much is in their system.
/:cl:

I'm having fun lee me alon

### CODE REVIEW

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 1]
```diff
@@ -394,3 +394,34 @@
 		M.Dizzy(10)
 		M.adjustStaminaLoss(5*REM, 0)
 	..()
+
+/datum/reagent/toxin/screech/
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: remove /

Comment (PopNotes) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: my b

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 2]
```diff
@@ -394,3 +394,34 @@
...
+	description = "Stimulates the vocal cords heavily, inducing involuntary yelling."
+	reagent_state = LIQUID
+	color = "#853358"
+	taste_description = "sour"
+	
+/datum/reagent/toxin/screech/on_mob_life(mob/living/M)
+	switch(current_cycle)
+		if(1 to 10)
+			if(prob(10))
+			M.emote("scream")
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: indentation is borked

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 3]
```diff
@@ -394,3 +394,34 @@
...
+		if(10 to 20)
+			if(prob(25))
+			M.emote("scream")
+			. = 1
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: why is this here?

**Comment by banthebantz** [comment]
REE WHY DON'T YOU PEOPLE CHECK THAT YOUR SHIT EVEN COMPILES

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 3 above)
Comment (PopNotes) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: just following standards ma'am

**Comment by PopNotes** [comment]
but it does compile :^)

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 3 above)
Comment (KayeArray) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: what standard?

Comment (PopNotes) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: It's good etiquette, apparently

I mean if it's superfluous then I'll remove it, I'm just going off memory.

**Comment by McBawbaggings** [comment]
change it so you have var/chance
var/chance is changed by the switch statement
if(chance)
    scream
    . = 1

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 4]
```diff
@@ -394,3 +394,35 @@
...
+	color = "#853358"
+	taste_description = "sour"
+	
+/datum/reagent/toxin/screech/on_mob_life(mob/living/M)
+	switch(current_cycle)
+		if(1 to 10)
+			if(prob(10))
+				M.emote("scream")
+				. = 1
+		if(10 to 20)
```
Comment (ghost) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: should be 11 to 20

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 5]
```diff
@@ -394,3 +394,35 @@
...
+			if(prob(25))
+				M.emote("scream")
+				. = 1
+		if(20 to 30)
```
Comment (ghost) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: 21 to 30

**Comment by PopNotes** [comment]
> change it so you have var/chance
>var/chance is changed by the switch statement
>if(chance)
>scream
>. = 1

how would this be better than what I already have tho

**Comment by McBawbaggings** [comment]
it stops you repeating the same block of M.emote(scream) and . = 1 4 times

**Comment by McBawbaggings** [comment]
```
/datum/reagent/toxin/screech/on_mob_life(mob/living/M)
	var/chance = 10
	switch(current_cycle)
		if(11 to 20)
			chance = 25
		if(21 to 30)
			chance = 40
		if(31 to 40)
			chance = 55
		if(41 to INFINITY)
			chance = 80
	if(chance)
		M.emote("scream")
		. = 1
	..()```
compared to
```
/datum/reagent/toxin/screech/on_mob_life(mob/living/M)
	switch(current_cycle)
		if(1 to 10)
			if(prob(10))
				M.emote("scream")
				. = 1
		if(11 to 20)
			if(prob(25))
				M.emote("scream")
				. = 1
		if(21 to 30)
			if(prob(40))
				M.emote("scream")
				. = 1
		if(31 to 40)
			if(prob(55))
				M.emote("scream")
				. = 1
		if(41 to INFINITY)
			if(prob(80))
				M.emote("scream")
				. = 1
	..()```

**Comment by PopNotes** [comment]
fair enough

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 6]
```diff
@@ -404,25 +404,17 @@
...
-			if(prob(55))
-				M.emote("scream")
-				. = 1
+			chance = 55
 		if(41 to INFINITY)
-			if(prob(80))
-				M.emote("scream")
-				. = 1
+			chance = 80
+	if(chance)
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: prob(chance)

**Comment by McBawbaggings** [comment]
yeah my bad, i forgot to add that

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 6 above)
Comment (PopNotes) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: shit broke

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 7]
```diff
@@ -414,7 +414,7 @@
 			chance = 55
 		if(41 to INFINITY)
 			chance = 80
-	if(chance)
+		prob(chance)
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: tut tut tut tut popnotes popnotes popnotes

you put the prob

IN THE IF

Comment (PopNotes) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: where the fuck do I put the prob

**Comment by McBawbaggings** [comment]
THE CODE IS INDENTED EVEN FURTHER

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 7 above)
Comment (KayeArray) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: IF(PROB(CHANCE))

LIKE THAT AND ONLY THAT

AND UNDO THE EXTRA INDENTATION NOW OK

**Comment by PopNotes** [comment]
I don't even know how that indent got there

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 8]
```diff
@@ -414,7 +414,7 @@
 			chance = 55
 		if(41 to INFINITY)
 			chance = 80
-		prob(chance)
+	if(prob(chance))
 		M.emote("scream")
 		. = 1
 	..()
```
Comment (ghost) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: wait this should be `return ..() || .`

Comment (KayeArray) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: no

**Comment by tortellinitony** [comment]
Add a check so that people will not suffocate

**Comment by PopNotes** [comment]
it's a fucking poison tony

**Comment by McBawbaggings** [comment]
like honestly, i think when i mad the flux scream PR it was at 30% per life tick and he died pretty much every game quickly

this is gonna kill people really quickly

**Comment by tortellinitony** [comment]
I don't support this then

**Comment by PopNotes** [comment]
Here's how to counter it: Epinephrine

File: hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 9]
```diff
@@ -394,3 +394,27 @@
...
 		M.adjustStaminaLoss(5*REM, 0)
 	..()
+
+/datum/reagent/toxin/screech
+	name = "Screechisol"
+	id = "screech"
+	description = "Stimulates the vocal cords heavily, inducing involuntary yelling."
+	reagent_state = LIQUID
+	color = "#853358"
+	taste_description = "sour"
```
Comment (ghost) [inline hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: wouldn't this be better as "salty"?

Comment (PopNotes) [reply hippiestation/code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: we'll reach a compromise
